pydantic/mypy.py

```diff
@@ -0,0 +1,624 @@
+                # This is the same logic used in the dataclasses plugin
+                continue
+
+            node = sym.node
+            if isinstance(node, PlaceholderNode):  # pragma: no cover
+                # See the PlaceholderNode docstring for more detail about how this can occur
+                # Basically, it is an edge case when dealing with complex import logic
+                # This is the same logic used in the dataclasses plugin
+                continue
+            assert isinstance(node, Var)
+
+            # x: ClassVar[int] is ignored by dataclasses.
+            if node.is_classvar:
+                continue
+
+            is_required = self.get_is_required(cls, stmt, lhs)
+            alias, has_dynamic_alias = self.get_alias_info(stmt)
+            if (
+                has_dynamic_alias
+                and not model_config.allow_population_by_field_name
+                and not self.plugin_config.allow_required_dynamic_aliases
+            ):
+                error_required_dynamic_aliases(ctx.api, stmt)
+            fields.append(
+                PydanticModelField(
+                    name=lhs.name,
+                    is_required=is_required,
+                    alias=alias,
+                    has_dynamic_alias=has_dynamic_alias,
+                    line=stmt.line,
+                    column=stmt.column,
+                )
+            )
+            known_fields.add(lhs.name)
+        all_attrs = fields.copy()
+        for info in cls.info.mro[1:]:  # 0 is the current class, -2 is BaseModel, -1 is object
+            if METADATA_KEY not in info.metadata:
+                continue
+
+            super_attrs = []
+            # Each class depends on the set of attributes in its dataclass ancestors.
+            ctx.api.add_plugin_dependency(make_wildcard_trigger(info.fullname()))
+
+            for name, data in info.metadata[METADATA_KEY]['attributes'].items():
+                if name not in known_fields:
+                    attr = PydanticModelField.deserialize(info, data)
+                    known_fields.add(name)
+                    super_attrs.append(attr)
+                else:
+                    (attr,) = [a for a in all_attrs if a.name == name]
+                    all_attrs.remove(attr)
+                    super_attrs.append(attr)
+            all_attrs = super_attrs + all_attrs
+        return all_attrs
+
+    def is_settings(self) -> bool:
+        for info in self._ctx.cls.info.mro[:-1]:  # 0 is the current class, -1 is object
+            if info.fullname() == BASESETTINGS_FULLNAME:
+                return True
+        return False
+
+    def add_initializer(self, fields: List['PydanticModelField'], config: 'ModelConfigData', is_settings: bool) -> None:
+        ctx = self._ctx
+        typed = self.plugin_config.init_typed
+        use_alias = config.allow_population_by_field_name is not True
```

Comment:
In general it is not possible to statically determine the truthiness of the config settings, but even to handle a cases like `= 1` would add a painful amount of additional code. So I actually set things up so that you get a separate error if you set the config to anything that isn't a boolean.

I originally wrote the config to have `Optional[bool]` values so I could tell whether they were set, but I ultimately made it always bool. So I think I can safely change the `is not True` to `is False`.

Reply:
Okay, this is good.